src/features/ai/sidebar/RewriteWidget.tsx

```diff
@@ -0,0 +1,148 @@
+import { useQuery } from '@tanstack/react-query';
+import { useCallback, useEffect, useState } from 'react';
+import { VscChevronLeft, VscChevronRight } from 'react-icons/vsc';
+
+import { askForRewrite } from '../../../api/rewrite';
+import { REWRITE_MANNER, RewriteOptions } from '../../../api/rewrite.config';
+import { emitEvent } from '../../../events';
+import { cx } from '../../../lib/cx';
+import { getCachedSetting } from '../../../settings/settingsManager';
+
+export function RewriteWidget({
+  selection,
+  className,
+  onChoiceSelected,
+}: {
+  selection: string;
+  className?: string;
+  onChoiceSelected?: (choice: string) => void;
+}) {
+  const openAiSettings = getCachedSetting('openAI');
+
+  const [rewriteOptions, setRewriteOptions] = useState<RewriteOptions>({
+    nChoices: 3,
+    manner: openAiSettings.manner,
+    temperature: openAiSettings.temperature,
+  });
+
+  useEffect(() => {
+    setRewriteOptions((state) => ({
+      ...state,
+      manner: openAiSettings.manner,
+      temperature: openAiSettings.temperature,
+    }));
+  }, [openAiSettings.manner, openAiSettings.temperature]);
+
+  const {
+    error,
+    data: rewrite,
+    refetch,
+    isFetching,
+  } = useQuery({
+    queryKey: [selection, rewriteOptions],
+    enabled: false,
+    queryFn: () => askForRewrite(selection, rewriteOptions),
+  });
+
+  const [selectedChoiceIndex, setSelectedChoiceIndex] = useState(0);
```

Comment:
You should add a `useEffect` to reset the `selectedChoiceIndex` state to 0 when rewrite is updated. Otherwise, if you go the third option, then select another manner and ask for rewrite again, you will stay on the third option, which feels a bit weird.